metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -297,20 +448,23 @@ def _populate_view_downstream_lineage(self, conn: SnowflakeConnection) -> None:
                 view_name: str = self.get_dataset_identifier_from_qualified_name(
                     db_row["VIEW_NAME"]
                 )
+                downstream_table: str = self.get_dataset_identifier_from_qualified_name(
+                    db_row["DOWNSTREAM_TABLE_NAME"]
+                )
                 if not self._is_dataset_pattern_allowed(
                     view_name, db_row["VIEW_DOMAIN"]
+                ) or not self._is_dataset_pattern_allowed(
+                    downstream_table, db_row["DOWNSTREAM_TABLE_DOMAIN"]
                 ):
                     continue
-                downstream_table: str = self.get_dataset_identifier_from_qualified_name(
-                    db_row["DOWNSTREAM_TABLE_NAME"]
-                )
+
                 # Capture view->downstream table lineage.
-                self._lineage_map[downstream_table].append(
+                self._lineage_map[downstream_table].update_lineage(
```

Comment:
are these calls protected from exceptions being thrown in parsing etc?

Reply:
yes, especially the part that's likely to cause exception (parsing dictionary to pydantic class) is protected from exceptions using try..except.